File: src/main/java/qmech/lib/objects/FluidBase.java
package qmech.lib.objects;

import net.minecraft.block.material.Material;
import net.minecraftforge.fluids.Fluid;
import net.minecraftforge.fluids.FluidRegistry;
import qmech.lib.util.LoggingHelper;

/**
 * Created by anshuman on 17-05-2014.
 */
public class FluidBase extends Fluid {
    public static FluidBase config (FluidBase fluid, int temp, int density, int viscosity, int luminescence) {
        LoggingHelper.getInstance().debug(String.format("Configuring Fluid (%s) with : \n" +
                        "temperature = %s \n" +
                        "density = %s \n" +
                        "viscosity = %s \n" +
                        "luminosity = %s",
                fluid.fluidName,
                temp, density, viscosity, luminescence
        ));

        fluid.temperature = temp;
        fluid.density = density;
        fluid.viscosity =viscosity;
        fluid.luminosity = luminescence;

        if (density < 0) {
            fluid.isGaseous = true;
        }

        return fluid;
    }

    public static FluidBase config (FluidBase fluid, FluidInfo info) {
        return config(fluid, info.temp, info.density, info.viscosity, info.luminosity);
    }

    public FluidBase(String fluidName, Material material, CreativeTabBase ctab) {
        super(fluidName);
        FluidRegistry.registerFluid(this);
        this.block = new FluidBlockBase(this, material, ctab);

        LoggingHelper.getInstance().debug(String.format("Creating Basic Fluid : %s", fluidName));
    }

    public static class FluidInfo {
        public int luminosity = 0;
        public int viscosity = 1000;
        public int density = 1000;
        public int temp = 295;

        public FluidInfo() {}

        public FluidInfo(int luminosity, int viscosity, int density, int temp) {
            this.luminosity = luminosity;
            this.viscosity = viscosity;
            this.density = density;
            this.temp = temp;
        }
    }
}

File: src/main/scala/qmech/lib/objects/fluid/FluidBase.scala
package qmech.lib.objects.fluid

import net.minecraftforge.fluids.{FluidRegistry, Fluid}
import net.minecraft.block.material.Material
import qmech.lib.objects.info.FluidInfo
import qmech.lib.objects.CreativeTabBase

class FluidBase(fluidName: String, material: Material, ctab: CreativeTabBase) extends Fluid(fluidName) {
  FluidRegistry.registerFluid(this)
  this.block = new FluidBlockBase(this, material, ctab)

  def this(fluidName: String, material: Material, ctab: CreativeTabBase, temp: Int, density: Int, viscosity: Int, luminescence: Int) = {
    this(fluidName, material, ctab)
    this setTemperature temp
    this setDensity density
    this setViscosity viscosity
    this setLuminosity luminescence
    isGaseous = density < 0
  }

  def this(fluidName: String, material: Material, ctab: CreativeTabBase, info: FluidInfo) = this(fluidName, material, ctab, info.temperature, info.density, info.viscosity, info.luminosity)
}
